rclpy/rclpy/context.py

```diff
@@ -149,3 +152,18 @@ def get_domain_id(self):
             raise RuntimeError('Context must be initialized before it can have a domain id')
         with self.__context, self._lock:
             return self.__context.get_domain_id()
+
+    def __enter__(self) -> "Context":
+        # We do not accept parameters here. If one wants to customize the init() call,
+        # they would have to call it manaully and not use the ContextManager convenience
+        self.init()
+        return self
+
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_val: BaseException | None,
+        exc_tb: TracebackType | None,
+    ) -> Literal[False]:
+        self.try_shutdown()
+        return False  # Reraise possible exceptions
```

Comment:
Nit, IIUC no need for a return statement. Returning `False` looks to be the the same as the default return of `None`. https://docs.python.org/3/reference/datamodel.html#object.__exit__